src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.h

```diff
@@ -110,6 +113,22 @@ class ReplicatedMergeTreePartCheckThread
     std::mutex start_stop_mutex;
     std::atomic<bool> need_stop { false };
     BackgroundSchedulePool::TaskHolder task;
+
+    // Background part check:
+    // (1) part for background check should be chosen from active parts but time to execute the check is chosen upfront
+    //     Therefore, background check is scheduled by enqueueing part check with empty part name,
+    //     so the part to check will be chosen among currently active parts at check time
+    // (2) after candidate for background check is chosen, we check latest time the part was checked.
+    //     If the part was checked recently (see background_part_check_delay_seconds), try to chose another part
```

Comment:
Need to update this comment, - it's incorrect. Currently, if the chosen part was checked recently then nothing is done in the current iteration